commands/apps.go

```diff
@@ -293,49 +305,74 @@ func (c *AppsInitCmd) InitializeNodeBasicApp(stdout, stderr bytes.Buffer) (err e
 	checkPkgJSON, err := os.Open("package.json")
 	if err != nil {
 		log.Println("[WARN] package.json does not exist. Creating package.json")
-		cmd := exec.Command("npm", "init", "-y")
-		cmd.Stdout = &stdout
-		cmd.Stderr = &stderr
-		err := cmd.Run()
+		err := c.CreatePkgJson(stdout, stderr)
 		if err != nil {
 			log.Println("[ERROR] There was an error creating package.json. Is node installed?")
-			panic(err)
+			return fmt.Errorf("there was an error creating package.json. Is node installed? %w", err)
 		} else {
 			log.Println("[INFO] package.json created")
 		}
 	}
 	defer checkPkgJSON.Close()
 	validPkgJSON, err := os.OpenFile("package.json", os.O_RDWR, 0777)
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("failed to open package.json %w", err)
 	}
+	defer validPkgJSON.Close()
 	log.Println("[INFO] Validating package.json")
-	finfo, err := validPkgJSON.Stat()
+	buf, err := ioutil.ReadFile("package.json")
 	if err != nil {
-		panic(err)
+		return fmt.Errorf("failed to read package.json %w", err)
 	}
-	buf := make([]byte, finfo.Size())
-	_, err = validPkgJSON.Read(buf)
+	fStr := string(buf)
+	if len(fStr) == 0 {
+		err := os.Remove("package.json")
+		if err != nil {
+			log.Println("[ERROR] unable to remove empty package.json")
+		}
+		log.Println("[WARN] package.json is empty. Creating package.json")
+		err = c.CreatePkgJson(stdout, stderr)
+		if err != nil {
+			log.Println("[ERROR] There was an error creating package.json. Is node installed?")
+			return fmt.Errorf("there was an error creating package.json. Is node installed? %w", err)
+		}
+		log.Println("[INFO] package.json created from empty file")
+		buf, err = ioutil.ReadFile("package.json")
+		if err != nil {
+			return fmt.Errorf("failed to read package.json %w", err)
+		}
+		fStr = string(buf)
+	}
+	jsonMap := make(map[string]interface{})
+	err = json.Unmarshal(buf, &jsonMap)
 	if err != nil {
-		panic(err)
+		log.Println("[ERROR] JSON format invalid for package.json")
+		return fmt.Errorf("package.json is not valid JSON %w", err)
 	}
-	fStr := string(buf)
-	if !strings.Contains(fStr, `"start": "`) {
-		replace := `"scripts": {
-    "start": "node YOUR_SERVER_HERE.js",`
-		fStr = strings.Replace(fStr, `"scripts": {`, replace, 1)
-		err = validPkgJSON.Truncate(0)
+	lv := jsonMap["scripts"]
+	stringMap, ok := lv.(map[string]interface{})
```

Comment:
Is there a better explanation than `stringMap`? Like files, lines, packages, etc?